File: find-n-unique-integers-sum-upto-zero/find-n-unique-integers-sum-upto-zero.py
class Solution(object):
    def sumZero(self, n):
        isEven = n % 2 == 0
        left = 0 
        num = -(n // 2) 
        res = []

        while left < n:
            res.append(num)
            if num == -1 and isEven:
                num += 2 
            else:
                num += 1 
            left += 1 

        return res  
        
        
        

File: find-n-unique-integers-sum-upto-zero/find-n-unique-integers-sum-upto-zero.cpp
class Solution
{
public:
    vector<int> sumZero(int n)
    {
        int left = 0;
        int num = -floor((double)n / 2);
        vector<int> res;

        while (left < n)
        {
            res.push_back(num);
            if (num == -1 && n % 2 == 0)
            {
                num += 2;
            }
            else
            {
                num += 1;
            }
            left += 1;
        }

        return res;
    }
};